packages/forms/src/Components/BaseFileUpload.php

```diff
@@ -104,6 +106,12 @@ protected function setUp(): void
         });
 
         $this->saveUploadedFileUsing(function (BaseFileUpload $component, TemporaryUploadedFile $file): string {
+            if ($this->getKeepFilename()) {
```

Comment:
Instead of duplicating the code here, should we use `$component->getKeepFilename() ? $file->getClientOriginalName() : $file->getFilename()`?

Also, all instances of `$this` inside closures need to be replaced with `$component` to ensure that repeaters aren't broken with this new functionality.

Reply:
Fixed, if I understand correctly